### ISSUE TITLE
[WC 2] no confirmation screen for signTransaction and signTransactions works well

### ISSUE THREAD

**Issue by juliancwirko** [ElrondNetwork/elrond-sdk-erdjs-wallet-connect-provider#11]
I think there is something wrong with signing a single transaction in V2. When I try to use `signTransaction` it doesn't show the confirmation screen in the Maiar app, but when I use `signTransactions`, it works well. Not sure what is a cause, but it waits for confirmation in https://github.com/ElrondNetwork/elrond-sdk-erdjs-wallet-connect-provider/blob/providerV2/src/walletConnectProviderV2.ts#L351.

**Comment by andreibancioiu** [comment]
Hello @juliancwirko, thank you for reporting the issue :+1:  

Do you think you can provide extra details about the Maiar app version? Did you test on Android or on iOS? If you've tested on Android, did you use the latest version, `1.5.21`?

I assume the `erdjs-wallet-connect-provider` is not the [latest stable](https://github.com/ElrondNetwork/elrond-sdk-erdjs-wallet-connect-provider/releases/tag/v2.0.3) release, but a beta instead, since you use V2. Is my assumption correct?

Thank you :pray:

**Comment by andreibancioiu** [comment]
We are currently investigating the issue and we'll come back with a fix.

:pray:

**Comment by juliancwirko** [comment]
Hey, it is `Maiar 1.5.20` Android and ` "@elrondnetwork/erdjs-wallet-connect-provider": "^2.1.0-beta.2",`
Thanks!

**Comment by andreibancioiu** [comment]
@juliancwirko, thank you :pray: 

The change is going to be applied on Maiar (Android), so that support for `signTransaction` (the singular version) is enabled, as well.

**Comment by lcswillems** [comment]
@andreibancioiu Are there been any progress on this side?

**Comment by andreibancioiu** [comment]
Support has been added in Maiar for Android `v1.5.40`, by @ditzdragos.

Thank you :pray:

**Comment by juliancwirko** [comment]
@andreibancioiu I haven't checked till now, and it seems that it still has some problems. Now with v1.5.40, the confirm/sign transaction screen shows up, but after confirmation, I got the following:

```
The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type object.
```
The transaction instance is the same as for the `signTransactions`. It looks like the type checks are incorrect somewhere, and it checks types for multiple transactions instead of one.

**Comment by lcswillems** [comment]
@juliancwirko @andreibancioiu Has this issue disappeared?

**Comment by juliancwirko** [comment]
I still have the same error with beta4 on Android and Maiar 1.5.44.

**Comment by lcswillems** [comment]
@andreibancioiu Maybe would it be good to reopen the issue?

**Comment by andreibancioiu** [comment]
Hello @lcswillems, hello @juliancwirko, 

I am re-opening the issue.

Thank you for the provided details, I will forward them.

**Comment by ditzdragos** [comment]
@juliancwirko thank you for the reproduction steps!
The fix will be available in Maiar 1.5.45

**Comment by ditzdragos** [comment]
Fix available in the latest store update: `1.5.46`

**Comment by juliancwirko** [comment]
I can confirm. It works now. Thanks!
